File: trunk/Firefly/Examples/Mapping/VB/SerializerExample/Src/DataModel.vb
'==========================================================================
'
'  File:        DataModel.vb
'  Location:    Firefly.Examples <Visual Basic .Net>
'  Description: 数据模型
'  Version:     2010.11.17.
'  Author:      F.R.C.
'  Copyright(C) Public Domain
'
'==========================================================================

Imports System
Imports System.Collections.Generic

'版本1的DataEntry
Public Class DataEntryVersion1
    Public Name As String
    Public Data As Byte()
End Class

'当前版本(版本2)的DataEntry
Public Class DataEntry
    Public Name As String
    Public Data As Byte()
    Public Attribute As String
End Class

Public Class ImmutableDataEntry(Of T)
    Public ReadOnly Property Name As String
        Get
            Return NameValue
        End Get
    End Property
    Public ReadOnly Property Data As T
        Get
            Return DataValue
        End Get
    End Property

    Private NameValue As String
    Private DataValue As T
    Public Sub New(ByVal Name As String, ByVal Data As T)
        Me.NameValue = Name
        Me.DataValue = Data
    End Sub
End Class

Public Class DataObject
    Public DataEntries As New Dictionary(Of String, DataEntry)
    Public ImmutableDataEntries As New Dictionary(Of String, ImmutableDataEntry(Of Byte()))
End Class

File: trunk/Firefly/Examples/Mapping/C#/SerializerExample/Src/DataModel.cs
//==========================================================================
//
//  File:        DataModel.cs
//  Location:    Firefly.Examples <Visual Basic .Net>
//  Description: 数据模型
//  Version:     2010.11.17.
//  Author:      F.R.C.
//  Copyright(C) public Domain
//
//==========================================================================

using System;
using System.Collections.Generic;

//版本1的DataEntry
public class DataEntryVersion1
{
    public String Name;
    public Byte[] Data;
}

//当前版本(版本2)的DataEntry
public class DataEntry
{
    public String Name;
    public Byte[] Data;
    public String Attribute;
}

public class ImmutableDataEntry<T>
{
    public String Name
    {
        get
        {
            return NameValue;
        }
    }
    public T Data
    {
        get
        {
            return DataValue;
        }
    }

    private String NameValue;
    private T DataValue;
    public ImmutableDataEntry(String Name, T Data)
    {
        this.NameValue = Name;
        this.DataValue = Data;
    }
}

public class DataObject
{
    public Dictionary<String, DataEntry> DataEntries = new Dictionary<String, DataEntry>();
    public Dictionary<String, ImmutableDataEntry<Byte[]>> ImmutableDataEntries = new Dictionary<String, ImmutableDataEntry<Byte[]>>();
}
